lib/clientv3/ippool.go

```diff
@@ -63,9 +63,48 @@ func (r ipPools) Create(ctx context.Context, res *apiv3.IPPool, opts options.Set
 		return nil, err
 	}
 
+	// Check that there are no existing blocks in the pool range that have a different block size.
+	poolBlockSize := res.Spec.BlockSize
+	poolIP, poolCIDR, err := net.ParseCIDR(res.Spec.CIDR)
+	if err != nil {
+		return nil, cerrors.ErrorParsingDatastoreEntry{
+			RawKey:   "CIDR",
+			RawValue: string(res.Spec.CIDR),
+			Err:      err,
+		}
+	}
+
+	ipVersion := 4
+	if poolIP.To4() == nil {
+		ipVersion = 6
+	}
+
+
+	blocks, err := r.client.backend.List(ctx, model.BlockListOptions{IPVersion: ipVersion}, "")
+	if _, ok := err.(cerrors.ErrorOperationNotSupported); !ok && err != nil {
+		// There was an error and it wasn't OperationNotSupported - return it.
+		return nil, err
+	} else if err == nil {
+		// Skip the block check if the error is OperationUnsupported - IPAM is not supported on KDD.
+		for _, b := range blocks.KVPairs {
+			k := b.Key.(model.BlockKey)
+			ones, _ := k.CIDR.Mask.Size()
+			// Check if this block has a different size to the pool, and that it overlaps with the pool.
+			if ones != poolBlockSize && k.CIDR.IsNetOverlap(*poolCIDR) {
+				return nil, cerrors.ErrorValidation{
+					ErroredFields: []cerrors.ErroredField{{
+						Name:   "IPPool.Spec.BlockSize",
+						Reason: "IPPool BlocksSize conflicts with existing allocations that use a different blockSize",
```

Comment:
BlocksSize -> blockSize (to be consistent with name as present in the user-facing yaml/json)